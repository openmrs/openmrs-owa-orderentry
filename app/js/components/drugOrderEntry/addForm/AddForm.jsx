import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import FreeText from './FreeText';
import StandardDose from './StandardDose';
import DosageTabs from '../../tabs/DosageTabs';
import DosageTab from '../../tabs/DosageTab';
import getOrderEntryConfigurations from '../../../actions/orderEntryActions';
import { addDraftOrder } from '../../../actions/draftTableAction';
import { selectDrugSuccess } from '../../../actions/drug';
import { setOrderAction, setSelectedOrder } from '../../../actions/orderAction';

export class AddForm extends React.Component {
  state = {
    action: 'NEW',
    fields: {
      dose: '',
      dosingUnit: '',
      frequency: '',
      route: '',
      duration: '',
      durationUnit: '',
      dispensingQuantity: '',
      dispensingUnit: '',
      reason: '',
      drugInstructions: '',
    },
    previousOrder: null,
    fieldErrors: {
    },
    draftOrder: {},
    orders: [],
    orderNumber: 0,
    formType: 'Standard Dosage',
    dosingType: '',
    activeTabIndex: 0,
    options: {
      dosingUnit: [],
      frequency: [],
      route: [],
      durationUnit: [],
      dispensingUnit: [],
    },
  };

  componentDidMount() {
    this.props.getOrderEntryConfigurations();
  }

  componentDidUpdate(prevProps) {
    return (
      Object.keys(this.props.draftOrder).length ||
      Object.keys(this.props.editOrder).length
    ) && this.populateEditOrderForm();
  }

  handleFormTabs = (tabIndex) => {
    this.setState({
      activeTabIndex: tabIndex,
      fieldErrors: {},
    });
  }

  handleFormType = (formType) => {
    this.setState({ formType: formType.trim() });
  }

  handleSubmitDrugForm = () => {
    const {
      dose,
      dosingUnit,
      frequency,
      route,
      duration,
      durationUnit,
      dispensingQuantity,
      dispensingUnit,
      reason,
      drugInstructions,
    } = this.state.fields;

    this.setState({
      draftOrder: {
        drugName: this.props.drugName,
        action: this.state.action,
        dose,
        dosingUnit,
        frequency,
        route,
        duration,
        durationUnit,
        dispensingUnit,
        dispensingQuantity,
        reason,
        previousOrder: this.state.previousOrder,
        drugInstructions,
        careSetting: this.props.careSetting.uuid,
        drug: this.props.drugUuid,
        orderer: this.props.session.currentProvider.uuid,
        dosingType: this.state.formType === 'Standard Dosage' ?
          'org.openmrs.SimpleDosingInstructions' :
          'org.openmrs.FreeTextDosingInstructions',
        type: "drugorder",
        orderNumber: (this.state.action === 'NEW') ? this.props.draftOrders.length :
          this.props.orderNumber,
      },
    }, () => {
      this.props.addDraftOrder(this.state.draftOrder);
      if (this.state.draftOrder.action === 'NEW') {
        this.props.setOrderAction('DRAFT', this.state.orderNumber);
      } else {
        this.props.setOrderAction('DRAFT_EDIT', this.props.orderNumber);
      }
    });
    this.props.selectDrugSuccess('');
    this.clearDrugForms();
    this.props.clearSearchField();
    if (this.props.activity === 'DRAFT_ORDER_EDIT') {
      this.props.setSelectedOrder({
        currentOrderType: this.props.currentOrderType,
        selectedOrder: null,
        activity: null,
      });
    }
  }


  /**
   * Validation of datalist tag values using onblur event handler
   */
  handleValidation = (event) => {
    const {
      drugDosingUnits, orderFrequencies, drugRoutes, durationUnits, drugDispensingUnits,
    } = this.props.allConfigurations;
    const { name, value } = event.target;
    let item = false;
    const options = [];
    switch (name) {
      case 'dosingUnit':
        item = drugDosingUnits.filter(unit => unit.display === value).length > 0;
        break;
      case 'frequency':
        item = orderFrequencies.filter(frequency => frequency.display === value).length > 0;
        break;
      case 'route':
        item = drugRoutes.filter(route => route.display === value).length > 0;
        break;
      case 'durationUnit':
        item = durationUnits.filter(unit => unit.display === value).length > 0;
        break;
      case 'dispensingUnit':
        item = drugDispensingUnits.filter(unit => unit.display === value).length > 0;
        break;
      default:
        item = value.length > 0;
    }

    { this.setState({
      ...this.state,
      fields: {
        ...this.state.fields, [name]: item ? value : '',
      },
      fieldErrors: {
        ...this.state.fieldErrors, [name]: !item,
      },
      options: {
        ...this.state.options, [name]: options,
      },
    }); }
  }

  handleChange = (event) => {
    const { name, value } = event.target;
    const {
      drugDosingUnits, orderFrequencies, drugRoutes, durationUnits, drugDispensingUnits,
    } = this.props.allConfigurations;
    let options;
    if (value) {
      switch (name) {
        case 'dosingUnit':
          options = drugDosingUnits.filter(unit => (
            unit.display.toLowerCase().indexOf(value.toLowerCase()) === 0));
          break;
        case 'frequency':
          options = orderFrequencies.filter(frequency => (
            frequency.display.toLowerCase().indexOf(value.toLowerCase()) === 0));
          break;
        case 'route':
          options = drugRoutes.filter(route => (
            route.display.toLowerCase().indexOf(value.toLowerCase()) === 0));
          break;
        case 'dispensingUnit':
          options = drugDispensingUnits.filter(unit => (
            unit.display.toLowerCase().indexOf(value.toLowerCase()) === 0));
          break;
        case 'durationUnit':
          options = durationUnits.filter(unit => (
            unit.display.toLowerCase().indexOf(value.toLowerCase()) === 0));
          break;
        default:
          // does nothing
      }
    }

    this.setState({
      ...this.state,
      fields: { ...this.state.fields, [name]: value },
      options: { ...this.state.options, [name]: options },
    });
  }

  handleCancel = () => {
    this.props.selectDrugSuccess('');
    this.clearDrugForms();
    this.props.clearSearchField();
    this.props.setOrderAction('DISCARD_ONE', this.props.orderNumber);
    this.props.setSelectedOrder({
      currentOrderType: this.props.currentOrderType,
      selectedOrder: null,
      activity: null,
    });
  }
  clearDrugForms = () => {
    this.setState({
      action: 'NEW',
      fields: {
        dose: '',
        dosingUnit: '',
        frequency: '',
        route: '',
        duration: '',
        durationUnit: '',
        dispensingQuantity: '',
        dispensingUnit: '',
        reason: '',
        drugInstructions: '',
      },
      previousOrder: null,
      fieldErrors: {
      },
      orderNumber: 0,
      formType: 'Standard Dosage',
      dosingType: '',
      activeTabIndex: 0,
      options: {
        dosingUnit: [],
        frequency: [],
        route: [],
        durationUnit: [],
        dispensingUnit: [],
      },
    });
  }

  activateSaveButton = () => {
    const {
      dose,
      dosingUnit,
      frequency,
      route,
      dispensingQuantity,
      dispensingUnit,
      duration,
      durationUnit,
      drugInstructions,
    } = this.state.fields;

    if (Object.values(this.state.fieldErrors).includes(true)) {
      return true;
    } else if (this.state.formType === 'Standard Dosage') {
      if ((duration && !durationUnit) ||
        !(
          dose &&
          dosingUnit &&
          frequency &&
          route &&
          dispensingQuantity &&
          dispensingUnit
        )) {
        return true;
      }
    } else if (this.state.formType === 'Free Text' &&
      !(drugInstructions && dispensingQuantity && dispensingUnit)) {
      return true;
    }
    return false;
  }

  populateEditOrderForm = () => {
    const { editOrder, draftOrder } = this.props;
    this.setState({
      activeTabIndex: (draftOrder.dosingType || editOrder.dosingType) === 'org.openmrs.SimpleDosingInstructions' ? 0 : 1,
      action: draftOrder.action || editOrder.action,
      previousOrder: editOrder.uuid || ((draftOrder.action === 'REVISE') && draftOrder.previousOrder) || null,
      formType: (draftOrder.dosingType || editOrder.dosingType) === 'org.openmrs.SimpleDosingInstructions' ? 'Standard Dosage' : 'Free Text',
      dosingType: draftOrder.dosingType || editOrder.dosingType,
      fields: {
        dose: editOrder.dose || draftOrder.dose || '',
        dosingUnit: (editOrder.doseUnits && editOrder.doseUnits.display) || draftOrder.dosingUnit || '',
        frequency: (editOrder.frequency && editOrder.frequency.display) || draftOrder.frequency || '',
        route: (editOrder.route && editOrder.route.display) || draftOrder.route || '',
        duration: editOrder.duration || draftOrder.duration || '',
        durationUnit: (editOrder.durationUnits && editOrder.durationUnits.display) || draftOrder.durationUnit || '',
        dispensingQuantity: editOrder.quantity || draftOrder.dispensingQuantity || '',
        dispensingUnit: (editOrder.quantityUnits && editOrder.quantityUnits.display) || draftOrder.dispensingUnit || '',
        reason: editOrder.asNeededCondition || draftOrder.reason || '',
        drugInstructions: editOrder.dosingInstructions || draftOrder.drugInstructions || '',
      },
    });
    this.props.removeOrder();
  }

  renderDrugOrderForms = () => (
    <div>
      <DosageTabs
        formType={this.handleFormType}
        activeTabIndex={this.state.activeTabIndex}
        handleFormTabs={this.handleFormTabs}
      >
        <DosageTab tabName="Standard Dosage &nbsp;" icon="icon-th-large">
          <StandardDose
            fields={this.state.fields}
            fieldErrors={this.state.fieldErrors}
            options={this.state.options}
            handleValidation={this.handleValidation}
            activateSaveButton={this.activateSaveButton}
            handleChange={this.handleChange}
            handleSubmit={this.handleSubmitDrugForm}
            handleCancel={this.handleCancel} />
          <br />
        </DosageTab>
        <DosageTab tabName="Free Text" icon="icon-edit">
          <FreeText
            fields={this.state.fields}
            fieldErrors={this.state.fieldErrors}
            options={this.state.options}
            handleValidation={this.handleValidation}
            activateSaveButton={this.activateSaveButton}
            handleChange={this.handleChange}
            handleCancel={this.handleCancel}
            handleSubmit={this.handleSubmitDrugForm}
          />
          <br />
        </DosageTab>
      </DosageTabs>
    </div>
  );

  render() {
    return (
      <div>
        { this.props.drugUuid && this.renderDrugOrderForms() }
      </div>
    );
  }
}

const mapStateToProps = ({
  orderEntryConfigurations,
  drugSearchReducer,
  draftReducer: { draftDrugOrders },
  openmrs: { session },
  orderSelectionReducer: { activity },
}) =>
  ({
    activity,
    drug: drugSearchReducer.selected,
    draftOrders: draftDrugOrders,
    allConfigurations: ((orderEntryConfigurations || {}).configurations || {}),
    session,
  });

AddForm.propTypes = {
  activity: PropTypes.string,
  clearSearchField: PropTypes.func.isRequired,
  currentOrderType: PropTypes.object.isRequired,
  selectDrugSuccess: PropTypes.func,
  getOrderEntryConfigurations: PropTypes.func,
  addDraftOrder: PropTypes.func.isRequired,
  setOrderAction: PropTypes.func.isRequired,
  orderNumber: PropTypes.number,
  draftOrders: PropTypes.arrayOf(PropTypes.any),
  allConfigurations: PropTypes.object.isRequired,
  drugName: PropTypes.string,
  careSetting: PropTypes.object.isRequired,
  drugUuid: PropTypes.string,
  removeOrder: PropTypes.func.isRequired,
  editOrder: PropTypes.shape({
    dose: PropTypes.number,
    doseUnits: PropTypes.shape({}),
    frequency: PropTypes.shape({}),
    route: PropTypes.shape({}),
    duration: PropTypes.number,
    durationUnits: PropTypes.shape({}),
    quantity: PropTypes.number,
    quantityUnits: PropTypes.shape({}),
    asNeededCondition: PropTypes.string,
    dosingInstructions: PropTypes.string,
    dosingType: PropTypes.string,
  }),
  draftOrder: PropTypes.shape({}),
  setSelectedOrder: PropTypes.func.isRequired,
  session: PropTypes.shape({
    currentProvider: PropTypes.shape({
      uuid: PropTypes.string,
    }),
  }).isRequired,
};

AddForm.defaultProps = {
  activity: '',
  selectDrugSuccess: {},
  getOrderEntryConfigurations: () => {},
  drugName: '',
  editOrder: {},
  draftOrder: {},
  draftOrders: [],
  drugUuid: '',
  orderNumber: '0',
};

export default connect(
  mapStateToProps,
  {
    getOrderEntryConfigurations,
    selectDrugSuccess,
    addDraftOrder,
    setOrderAction,
    setSelectedOrder,
  },
)(AddForm);
