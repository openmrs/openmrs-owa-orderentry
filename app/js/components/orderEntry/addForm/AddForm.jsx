import React from 'react';
import FreeText from './FreeText';
import StandardDose from './StandardDose';

class AddForm extends React.Component {

  render() {
    return (
      <div className="row">
        <div className="col-lg-11 col-sm-11 col-md-11 col-md-offset-1 col-sm-offset-1 col-lg-offset-1">
          <ul className="nav nav-pills mb-3" id="pills-tab" role="tablist">
            <li className="nav-item">
              <a className="nav-link"
                id="standard-tab"
                data-toggle="pill"
                href="#standard"
                role="tab"
                aria-controls="standard"
                aria-selected="false">Standard Dosing
              </a>
            </li>
            <li className="nav-item">
              <a className="nav-link"
                id="free-tab"
                data-toggle="pill"
                href="#free"
                role="tab"
                aria-controls="free"
                aria-selected="false">Free Text
              </a>
            </li>
          </ul>
          <br />

          <div className="tab-content" id="pills-tabContent">
            <FreeText />
            <StandardDose />
          </div>

        </div>
      </div>
    );
  }
}

export default AddForm;
