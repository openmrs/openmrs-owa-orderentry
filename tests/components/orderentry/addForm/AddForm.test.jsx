import React from 'react';

import AddForm from '../../../../app/js/components/orderEntry/addForm/AddForm';
import store from '../../../../app/js/redux-store';

describe('Test for Adding an order', () => {
  it('should render component', () => {
    const wrapper = shallow(<AddForm />  );
    expect(wrapper).toMatchSnapshot()
  });
});
