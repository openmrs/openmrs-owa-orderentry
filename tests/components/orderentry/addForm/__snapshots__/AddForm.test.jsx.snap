// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Adding an order should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="col-lg-11 col-sm-11 col-md-11 col-md-offset-1 col-sm-offset-1 col-lg-offset-1"
      >
        <ul
          className="nav nav-pills mb-3"
          id="pills-tab"
          role="tablist"
        >
          <li
            className="nav-item"
          >
            <a
              aria-controls="standard"
              aria-selected="false"
              className="nav-link"
              data-toggle="pill"
              href="#standard"
              id="standard-tab"
              role="tab"
            >
              Standard Dosing
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-controls="free"
              aria-selected="false"
              className="nav-link"
              data-toggle="pill"
              href="#free"
              id="free-tab"
              role="tab"
            >
              Free Text
            </a>
          </li>
        </ul>
        <br />
        <div
          className="tab-content"
          id="pills-tabContent"
        >
          <FreeText />
          <StandardDose />
        </div>
      </div>,
      "className": "row",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ul
            className="nav nav-pills mb-3"
            id="pills-tab"
            role="tablist"
          >
            <li
              className="nav-item"
            >
              <a
                aria-controls="standard"
                aria-selected="false"
                className="nav-link"
                data-toggle="pill"
                href="#standard"
                id="standard-tab"
                role="tab"
              >
                Standard Dosing
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-controls="free"
                aria-selected="false"
                className="nav-link"
                data-toggle="pill"
                href="#free"
                id="free-tab"
                role="tab"
              >
                Free Text
              </a>
            </li>
          </ul>,
          <br />,
          <div
            className="tab-content"
            id="pills-tabContent"
          >
            <FreeText />
            <StandardDose />
          </div>,
        ],
        "className": "col-lg-11 col-sm-11 col-md-11 col-md-offset-1 col-sm-offset-1 col-lg-offset-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="nav-item"
              >
                <a
                  aria-controls="standard"
                  aria-selected="false"
                  className="nav-link"
                  data-toggle="pill"
                  href="#standard"
                  id="standard-tab"
                  role="tab"
                >
                  Standard Dosing
                </a>
              </li>,
              <li
                className="nav-item"
              >
                <a
                  aria-controls="free"
                  aria-selected="false"
                  className="nav-link"
                  data-toggle="pill"
                  href="#free"
                  id="free-tab"
                  role="tab"
                >
                  Free Text
                </a>
              </li>,
            ],
            "className": "nav nav-pills mb-3",
            "id": "pills-tab",
            "role": "tablist",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  aria-controls="standard"
                  aria-selected="false"
                  className="nav-link"
                  data-toggle="pill"
                  href="#standard"
                  id="standard-tab"
                  role="tab"
                >
                  Standard Dosing
                </a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-controls": "standard",
                  "aria-selected": "false",
                  "children": "Standard Dosing",
                  "className": "nav-link",
                  "data-toggle": "pill",
                  "href": "#standard",
                  "id": "standard-tab",
                  "role": "tab",
                },
                "ref": null,
                "rendered": "Standard Dosing",
                "type": "a",
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  aria-controls="free"
                  aria-selected="false"
                  className="nav-link"
                  data-toggle="pill"
                  href="#free"
                  id="free-tab"
                  role="tab"
                >
                  Free Text
                </a>,
                "className": "nav-item",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-controls": "free",
                  "aria-selected": "false",
                  "children": "Free Text",
                  "className": "nav-link",
                  "data-toggle": "pill",
                  "href": "#free",
                  "id": "free-tab",
                  "role": "tab",
                },
                "ref": null,
                "rendered": "Free Text",
                "type": "a",
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <FreeText />,
              <StandardDose />,
            ],
            "className": "tab-content",
            "id": "pills-tabContent",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="col-lg-11 col-sm-11 col-md-11 col-md-offset-1 col-sm-offset-1 col-lg-offset-1"
        >
          <ul
            className="nav nav-pills mb-3"
            id="pills-tab"
            role="tablist"
          >
            <li
              className="nav-item"
            >
              <a
                aria-controls="standard"
                aria-selected="false"
                className="nav-link"
                data-toggle="pill"
                href="#standard"
                id="standard-tab"
                role="tab"
              >
                Standard Dosing
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-controls="free"
                aria-selected="false"
                className="nav-link"
                data-toggle="pill"
                href="#free"
                id="free-tab"
                role="tab"
              >
                Free Text
              </a>
            </li>
          </ul>
          <br />
          <div
            className="tab-content"
            id="pills-tabContent"
          >
            <FreeText />
            <StandardDose />
          </div>
        </div>,
        "className": "row",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul
              className="nav nav-pills mb-3"
              id="pills-tab"
              role="tablist"
            >
              <li
                className="nav-item"
              >
                <a
                  aria-controls="standard"
                  aria-selected="false"
                  className="nav-link"
                  data-toggle="pill"
                  href="#standard"
                  id="standard-tab"
                  role="tab"
                >
                  Standard Dosing
                </a>
              </li>
              <li
                className="nav-item"
              >
                <a
                  aria-controls="free"
                  aria-selected="false"
                  className="nav-link"
                  data-toggle="pill"
                  href="#free"
                  id="free-tab"
                  role="tab"
                >
                  Free Text
                </a>
              </li>
            </ul>,
            <br />,
            <div
              className="tab-content"
              id="pills-tabContent"
            >
              <FreeText />
              <StandardDose />
            </div>,
          ],
          "className": "col-lg-11 col-sm-11 col-md-11 col-md-offset-1 col-sm-offset-1 col-lg-offset-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li
                  className="nav-item"
                >
                  <a
                    aria-controls="standard"
                    aria-selected="false"
                    className="nav-link"
                    data-toggle="pill"
                    href="#standard"
                    id="standard-tab"
                    role="tab"
                  >
                    Standard Dosing
                  </a>
                </li>,
                <li
                  className="nav-item"
                >
                  <a
                    aria-controls="free"
                    aria-selected="false"
                    className="nav-link"
                    data-toggle="pill"
                    href="#free"
                    id="free-tab"
                    role="tab"
                  >
                    Free Text
                  </a>
                </li>,
              ],
              "className": "nav nav-pills mb-3",
              "id": "pills-tab",
              "role": "tablist",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    aria-controls="standard"
                    aria-selected="false"
                    className="nav-link"
                    data-toggle="pill"
                    href="#standard"
                    id="standard-tab"
                    role="tab"
                  >
                    Standard Dosing
                  </a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-controls": "standard",
                    "aria-selected": "false",
                    "children": "Standard Dosing",
                    "className": "nav-link",
                    "data-toggle": "pill",
                    "href": "#standard",
                    "id": "standard-tab",
                    "role": "tab",
                  },
                  "ref": null,
                  "rendered": "Standard Dosing",
                  "type": "a",
                },
                "type": "li",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    aria-controls="free"
                    aria-selected="false"
                    className="nav-link"
                    data-toggle="pill"
                    href="#free"
                    id="free-tab"
                    role="tab"
                  >
                    Free Text
                  </a>,
                  "className": "nav-item",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "aria-controls": "free",
                    "aria-selected": "false",
                    "children": "Free Text",
                    "className": "nav-link",
                    "data-toggle": "pill",
                    "href": "#free",
                    "id": "free-tab",
                    "role": "tab",
                  },
                  "ref": null,
                  "rendered": "Free Text",
                  "type": "a",
                },
                "type": "li",
              },
            ],
            "type": "ul",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <FreeText />,
                <StandardDose />,
              ],
              "className": "tab-content",
              "id": "pills-tabContent",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
